runtime/service/src/main/java/org/apache/polaris/service/catalog/generic/CatalogGenericTableEventServiceDelegator.java

```diff
@@ -78,7 +108,19 @@ public Response loadGenericTable(
       String genericTable,
       RealmContext realmContext,
       SecurityContext securityContext) {
-    return delegate.loadGenericTable(
-        prefix, namespace, genericTable, realmContext, securityContext);
+    String catalogName = getCatalogFromPrefix(prefix, realmContext);
+    polarisEventListener.onBeforeLoadGenericTable(
+        new CatalogGenericTableServiceEvents.BeforeLoadGenericTableEvent(
+            catalogName, namespace, genericTable));
+    Response resp =
+        delegate.loadGenericTable(prefix, namespace, genericTable, realmContext, securityContext);
+    polarisEventListener.onAfterLoadGenericTable(
+        new CatalogGenericTableServiceEvents.AfterLoadGenericTableEvent(
+            catalogName, namespace, ((LoadGenericTableResponse) resp.getEntity()).getTable()));
+    return resp;
+  }
+
+  private String getCatalogFromPrefix(String prefix, RealmContext realmContext) {
+    return prefixParser.prefixToCatalogName(realmContext, prefix);
```

Comment:
nit: inline (the method does nothing  more than `prefixToCatalogName`)

Reply:
Changed.